src/layers/basic.jl

```diff
@@ -211,48 +211,69 @@ function Base.show(io::IO, l::Diagonal)
 end
 
 """
-    Maxout(over)
+    Maxout(layers...)
+    Maxout(f, n_alts)
 
-The [Maxout](https://arxiv.org/abs/1302.4389) layer has a number of
-internal layers which all receive the same input. It returns the elementwise
-maximum of the internal layers' outputs.
+This contains a number of internal layes, each of which receives the same input.
+Its output is the elementwise maximum of the the internal layers' outputs.
 
-Maxout over linear dense layers satisfies the univeral approximation theorem.
-"""
-struct Maxout{FS<:Tuple}
-    over::FS
-end
+Instead of defining layers individually, you can provide a zero-argument function
+which constructs them, and the number to construct.
 
-"""
-    Maxout(f, n_alts)
+Maxout over linear dense layers satisfies the univeral approximation theorem.
+See Goodfellow, Warde-Farley, Mirza, Courville & Bengio "Maxout Networks" 
+[https://arxiv.org/abs/1302.4389](1302.4389).
 
-Construct a Maxout layer over `n_alts` instances of the layer given by `f`.
-The function takes no arguments and should return some callable layer.
-Conventionally, this is a linear dense layer.
+See also [`Parallel`](@ref) to reduce with other operators.
 
 # Examples
+```
+julia> m = Maxout(Dense([1;;], false, abs2), Dense([3;;]));
```

Comment:
I would favour using anonymous functions here to make it real explicit what each alternative is doing.

Reply:
This example is for the first signature, which accepts several layers. The second signature has an example below with `() ->` .

Reply:
I meant something like the following:
```suggestion
julia> m = Maxout(x -> abs2.(x), x -> x .* 3);
```
Otherwise new users will have to puzzle out what `[1;;]` means, what `Dense(::Array, false, ::Function)` does and that `Dense([1;;], false, abs2)(x) === abs2.(1 .* x .+ false) === abs2.(x)`

Reply:
Ah ok! That's clearer.